backend/packages/vesting-scan/src/service/vestingSummary.js

```diff
@@ -0,0 +1,93 @@
+const {
+  chain: { findBlockApi },
+  call,
+} = require("@osn/scan-common");
+
+const { getVestingEvents } = require("../store/event");
+const { getVestingCalls } = require("../store/call");
+
+async function generateVestingSummary(indexer) {
+  const events = getVestingEvents(indexer);
+  // For some blocks, events are not emitted.
+  // For example, in block 8730807, there is a mergeSchedules call but without any events.
+  const calls = getVestingCalls(indexer);
+  const accounts = await findChangedAccounts(events, calls);
+  if (accounts.length <= 0) {
+    return;
+  }
+
+  const api = await findBlockApi(indexer.blockHash);
+
+  const accountScheduels = await fetchAllVestings(api);
```

Comment:
It's not good we check all vestings in storage at every block there is vesting related actions. This will be a bottleneck for scan performance. Image we have 10k records in DB, and there is one account vesting action, I think we don't have to get all the storage just for one event/call.